python/interpret_community/mimic/models/linear_model.py

```diff
@@ -174,26 +171,240 @@ def predict_proba(self, dataset, **kwargs):
+        return _create_global_explanation(**kwargs)
+
+    def explain_local(self, evaluation_examples, **kwargs):
+        """Use LinearExplainer to get the local feature importances from the trained explainable model.
+
+        :param evaluation_examples: The evaluation examples to compute local feature importances for.
+        :type evaluation_examples: numpy or scipy array
+        :return: The local explanation of feature importances.
+        :rtype: Union[list, numpy.ndarray]
+        """
+        if self._linear_explainer is None:
+            self._linear_explainer = LinearExplainer(self._linear, (self.mean, self.covariance))
+        return self._linear_explainer.explain_local(evaluation_examples)
+
+    def predict(self, dataset, **kwargs):
+        """Call linear predict to predict labels using the glassbox model.
+
+        :param dataset: The dataset to predict on.
+        :type dataset: numpy or scipy array
+        :return: The predictions of the model.
+        :rtype: list
+        """
+        return super(LogisticRegression, self).predict(dataset, **kwargs)
+
+    def predict_proba(self, dataset, **kwargs):
+        """Call linear predict_proba to predict probabilities using the glassbox model.
+
+        :param dataset: The dataset to predict probabilities on.
+        :type dataset: numpy or scipy array
+        :return: The predictions of the model.
+        :rtype: list
+        """
+        return super(LogisticRegression, self).predict_proba(dataset, **kwargs)
+
+    __init__.__doc__ = (__init__.__doc__ +
+                        '\nUses the parameters for LogisticRegression:\n' +
+                        SKLogisticRegression.__doc__.replace('-', ''))
+
+    fit.__doc__ = (fit.__doc__ +
+                   '\nUses the parameters for LogisticRegression:\n' +
+                   SKLogisticRegression.fit.__doc__.replace('-', ''))
+
+    predict.__doc__ = (predict.__doc__ +
+                       '\nUses the parameters for LogisticRegression:\n' +
+                       SKLogisticRegression.predict.__doc__.replace('-', ''))
+
+    predict_proba.__doc__ = (predict_proba.__doc__ +
+                             '\nUses the parameters for LogisticRegression:\n' +
+                             SKLogisticRegression.predict_proba.__doc__.replace('-', ''))
+
+
+class LinearExplainableModel(LinearClassifierMixin, LinearBase, BaseExplainableModel):
+    available_explanations = [Extension.GLOBAL, Extension.LOCAL]
+    explainer_type = Extension.GLASSBOX
+
+    """Linear explainable model.
+
+    :param multiclass: Set to true to generate a multiclass model.
+    :type multiclass: bool
+    :param random_state: Int to seed the model.
+    :type random_state: int
+    """
+
+    def __init__(self, multiclass=False, random_state=DEFAULT_RANDOM_STATE, classification=True, **kwargs):
+        """Initialize the LinearExplainableModel.
+
+        :param multiclass: Set to true to generate a multiclass model.
```

Comment:
I'm in favor of requiring more user input for this stuff, but is this consistent with our experience for other parts of the system?

Reply:
And I think having a multiclass parameter and a classification parameter might be confusing to some users (maybe add more to the docstring to make the difference clear?)

Reply:
this is the explainable model, the API hasn't changed.  The user doesn't call the explainable model, they just pass it to mimic explainer and we initialize it.  This API hasn't actually changed, the code was just moved around in the diff.